File: server/src/B1ngo.Application.Features/Rooms/CreateRoom/CreateRoomCommand.cs
using B1ngo.Application.Common;
using B1ngo.Domain.Game;

namespace B1ngo.Application.Features.Rooms.CreateRoom;

public sealed record CreateRoomCommand(
    string HostDisplayName,
    int Season,
    string GrandPrixName,
    SessionType SessionType,
    int? MatrixSize = null,
    List<WinPatternType>? WinningPatterns = null) : ICommand<CreateRoomResponse>;

File: client/apps/b1ngo/src/app/core/api/models/commands/create-room.command.ts
import { SessionType } from '../types/session-type.type';
import { WinPatternType } from '../types/win-pattern-type.type';

export interface CreateRoomCommand {
  hostDisplayName: string;
  season: number;
  grandPrixName: string;
  sessionType: SessionType;
  matrixSize?: number;
  winningPatterns?: WinPatternType[];
}
